String.cxx

```diff
@@ -0,0 +1,184 @@
+/*
+ *  String.cxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <algorithm>
+#include <cctype>
+#include <charconv>
+#include <cstring>
+#include <iostream>
+#include <regex>
+#include <string>
+
+enum StringBaseConv {
+    BIN = 2,
+    DEC = 10,
+    HEX = 16,
+};
+
+enum class byte : unsigned char {};
```

Comment:
That's how `std::byte` is defined; the Arduino version is just `using byte = unsigned char;`

Reply:
Changes were made, thanks for pointing it out.